tools/hangar/__snapshots__/test_corpus/valid/structs.w_compile_tf-aws.md

```diff
@@ -1,5 +1,212 @@
 # [structs.w](../../../../../examples/tests/valid/structs.w) | compile | tf-aws
 
+## A.Struct.js
+```js
+module.exports = function(validator_req, fromInline) {
+  class A {
+    static _schema = {
+      id: "/A",
+      type: "object",
+      properties: {
+        field0: { type: "string" },
+      },
+      required: [
+        "field0",
+      ]
+    }
+    static _getDependencies() {
+      return {
+      }
+    }
+    static _getValidator() {
+      const v = new validator_req.Validator();
+      const addDependencies = ((deps) => { for (const [key, value] of Object.entries(deps)) {
+        v.addSchema(value, key);
+      }});
+      addDependencies(this._getDependencies());
+      return v;
+    }
+    static validate(obj) {
+      const result = this._getValidator().validate(obj, this._schema);
+      return result.errors;
+    }
+    static _toInflightType() {
+      return fromInline(`require("./A.Struct.js")(require("jsonschema"))`);
+    }
+  }
+  return A;
+};
+
+```
+
+## B.Struct.js
+```js
+module.exports = function(validator_req, fromInline) {
+  class B {
+    static _schema = {
+      id: "/B",
+      type: "object",
+      properties: {
+        ...require("./A.Struct.js")()._schema.properties,
+        field1: { type: "number" },
+        field2: { type: "string" },
+        field3: { "$ref": "/A" },
+      },
+      required: [
+        "field1",
+        "field2",
+        "field3",
+        ...require("./A.Struct.js")()._schema.required,
+      ]
+    }
+    static _getDependencies() {
+      return {
+        "/A": require("./A.Struct.js")()._schema,
+      }
+    }
+    static _getValidator() {
+      const v = new validator_req.Validator();
+      const addDependencies = ((deps) => { for (const [key, value] of Object.entries(deps)) {
+        v.addSchema(value, key);
+      }});
+      addDependencies(this._getDependencies());
+      addDependencies(require("./A.Struct.js")()._getDependencies());
+      return v;
+    }
+    static validate(obj) {
+      const result = this._getValidator().validate(obj, this._schema);
+      return result.errors;
```

Comment:
```suggestion
      return this._getValidator().validate(obj, this._schema).errors;
```

Reply:
Oh yea, originally I was doing some error message formatting here but now I can just oneline it.